### PR TITLE
Add the version number via setuptools_scm

### LINKED ISSUE #18: `pip check` isn't happy with latest release (3.1.0)
In builds with the latest `filelock`, the version number in the metadata is 0.0.0.  This is causing issues with `pip check`.

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=383996&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=e5c8ab1d-8ff9-5cae-b332-e15ae582ed2d&l=8961
```
virtualenv 20.4.7 has requirement filelock<4,>=3.0.0, but you have filelock 0.0.0.
```

I've seen this recently with another feedstock:
https://github.com/conda-forge/libcst-feedstock/issues/23
fixed in
https://github.com/conda-forge/libcst-feedstock/pull/24

I'll see if a similar fix will work here.  PR to come in a bit...

### PR DESCRIPTION
**Pull request by xylar**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

closes #18

### CODE REVIEW

**Comment by xylar** [comment]
It seems like this is largely an upstream problem (version is not defined anywhere in this release) so maybe this is just a temporary fix.

**Comment by xylar** [comment]
I tested locally and this worked:
```
$ grep -r "3\.1\.0"
site-packages/filelock-3.1.0.dist-info/RECORD:filelock-3.1.0.dist-info/INSTALLER,sha256=zuuue4knoyJ-UwPPXg8fezS7VCrXJQrAP7zeNuwvFQg,4
site-packages/filelock-3.1.0.dist-info/RECORD:filelock-3.1.0.dist-info/LICENSE,sha256=iNm062BXnBkew5HKBMFhMFctfu3EqG2qWL8oxuFMm80,1210
site-packages/filelock-3.1.0.dist-info/RECORD:filelock-3.1.0.dist-info/METADATA,sha256=DgMZ_ewv4edt85e0GyAuTuCyTwdqkFN6plEAmZh39Fk,5634
site-packages/filelock-3.1.0.dist-info/RECORD:filelock-3.1.0.dist-info/RECORD,,
site-packages/filelock-3.1.0.dist-info/RECORD:filelock-3.1.0.dist-info/REQUESTED,sha256=47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU,0
site-packages/filelock-3.1.0.dist-info/RECORD:filelock-3.1.0.dist-info/WHEEL,sha256=WzZ8cwjh8l0jtULNjYq1Hpr-WCqCRgPr--TX4P5I1Wo,110
site-packages/filelock-3.1.0.dist-info/RECORD:filelock-3.1.0.dist-info/direct_url.json,sha256=1u39L-H_nA228RmbXBOVtt2lopvNHl6ixj6go4ojrQ8,94
site-packages/filelock-3.1.0.dist-info/RECORD:filelock-3.1.0.dist-info/top_level.txt,sha256=NDrf9i5BNogz4hEdsr6Hi7Ws3TlSSKY4Q2Y9_-i2GwU,9
site-packages/filelock-3.1.0.dist-info/RECORD:filelock-3.1.0.dist-info/zip-safe,sha256=AbpHGcgLb-kRsJGnwFEktk7uzpZOCcBY74-YBdrKVGs,1
site-packages/filelock-3.1.0.dist-info/METADATA:Version: 3.1.0
site-packages/filelock/version.py:version = '3.1.0'
info/paths.json:      "_path": "site-packages/filelock-3.1.0.dist-info/INSTALLER",
info/paths.json:      "_path": "site-packages/filelock-3.1.0.dist-info/LICENSE",
info/paths.json:      "_path": "site-packages/filelock-3.1.0.dist-info/METADATA",
info/paths.json:      "_path": "site-packages/filelock-3.1.0.dist-info/RECORD",
info/paths.json:      "_path": "site-packages/filelock-3.1.0.dist-info/REQUESTED",
info/paths.json:      "_path": "site-packages/filelock-3.1.0.dist-info/WHEEL",
info/paths.json:      "_path": "site-packages/filelock-3.1.0.dist-info/direct_url.json",
info/paths.json:      "_path": "site-packages/filelock-3.1.0.dist-info/top_level.txt",
info/paths.json:      "_path": "site-packages/filelock-3.1.0.dist-info/zip-safe",
info/recipe/meta.yaml.template:{% set version = "3.1.0" %}
info/recipe/meta.yaml:  version: 3.1.0
info/recipe/meta.yaml:  fn: filelock-3.1.0.tar.gz
info/recipe/meta.yaml:  url: https://pypi.io/packages/source/f/filelock/filelock-3.1.0.tar.gz
info/recipe/meta.yaml:    - export SETUPTOOLS_SCM_PRETEND_VERSION=3.1.0
info/files:site-packages/filelock-3.1.0.dist-info/INSTALLER
info/files:site-packages/filelock-3.1.0.dist-info/LICENSE
info/files:site-packages/filelock-3.1.0.dist-info/METADATA
info/files:site-packages/filelock-3.1.0.dist-info/RECORD
info/files:site-packages/filelock-3.1.0.dist-info/REQUESTED
info/files:site-packages/filelock-3.1.0.dist-info/WHEEL
info/files:site-packages/filelock-3.1.0.dist-info/direct_url.json
info/files:site-packages/filelock-3.1.0.dist-info/top_level.txt
info/files:site-packages/filelock-3.1.0.dist-info/zip-safe
info/index.json:  "version": "3.1.0"
```

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

**Comment by xylar** [comment]
@jakirkham, @mariusvniekerk and @nicoddemus, I think this is ready to review.  Please take a look when you have a chance.

**Comment by xylar** [comment]
Needed for https://github.com/conda-forge/google-ads-feedstock/pull/9

File: recipe/meta.yaml [hunk 1]
```diff
@@ -12,13 +12,14 @@ source:
...
   noarch: python
-  number: 0
+  number: 1
   script: {{ PYTHON }} -m pip install --no-deps .
 
 requirements:
   host:
     - pip
     - python >=3.6
+    - setuptools_scm
```
Comment (xylar) [inline recipe/meta.yaml:22]: Just adding `setuptools_scm` as a dependency seems to fix things, as suggested in https://github.com/conda-forge/filelock-feedstock/issues/18#issuecomment-930521785

**Comment by xylar** [comment]
Okay, I modified things as @nicoddemus suggested in https://github.com/conda-forge/filelock-feedstock/issues/18#issuecomment-930521785.  That seemed to work great in a local test.  I think this is ready to merge once CI completes, unless someone has further suggestions.

**Comment by nicoddemus** [comment]
Yep looks good, thanks!

**Comment by xylar** [comment]
@nicoddemus, thanks a lot!  That got my other update unstuck, so it definitely seems to have worked.

**Comment by nicoddemus** [comment]
Awesome, thanks again for fixing it!

**Pull request merged**